File: server/models/Document.js
import mongoose from 'mongoose'

const documentSchema = mongoose.Schema({
  uid: {
    required: true,
    type: String
  },
  createdAt: {
    required: true,
    type: Number
  },
  title: {
    required: true,
    type: String,
    trim: true
  },
  content: {
    default: [],
    type: Array
  }
})

export const Document = mongoose.model('document', documentSchema)

File: lib/models/document.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'dart:convert';

class DocumentModel {
  final String uid;
  final String title;
  final List content;
  final DateTime createdAt;
  final String id;

  DocumentModel({
    required this.uid,
    required this.title,
    required this.content,
    required this.createdAt,
    required this.id,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'uid': uid,
      'title': title,
      'content': content,
      'createdAt': createdAt.millisecondsSinceEpoch,
      'id': id,
    };
  }

  factory DocumentModel.fromMap(Map<String, dynamic> map) {
    return DocumentModel(
      uid: map['uid'] as String,
      title: map['title'] as String,
      content: List.from((map['content'] as List)),
      createdAt: DateTime.fromMillisecondsSinceEpoch(map['createdAt']),
      id: map['_id'] as String,
    );
  }

  String toJson() => json.encode(toMap());

  factory DocumentModel.fromJson(String source) =>
      DocumentModel.fromMap(json.decode(source) as Map<String, dynamic>);
}
